Sources/Utils/Backtrace.cpp

```diff
@@ -22,20 +22,54 @@
 #include <sstream>
 #endif
 
+#if defined(ANDROID)
+// Android provides libunwind in the NDK instead of glibc backtrace
+#include <unwind.h>
+#endif
+
 namespace ds2 {
 namespace Utils {
 
-#if defined(OS_DARWIN) || (defined(__GLIBC__) && !defined(PLATFORM_TIZEN))
+#if defined(ANDROID)
+struct unwind_state {
+    void **buffer_cursor;
+    void **buffer_end;
+};
+
+static _Unwind_Reason_Code UnwindTrace(struct _Unwind_Context* context,
+                                       void* arg) {
+  struct unwind_state* state = reinterpret_cast<struct unwind_state*>(arg);
+  uintptr_t ip = _Unwind_GetIP(context);
+  if (ip != 0) {
+    if (state->buffer_cursor >= state->buffer_end)
+      return _URC_END_OF_STACK; // Backtrace will be truncated
+
+    *(state->buffer_cursor++) = reinterpret_cast<void*>(ip);
+  }
+  return _URC_NO_REASON;
+}
+
+// An approximation of glibc's backtrace using _Unwind_Backtrace
+static int backtrace(void *stack_buffer[], int stack_size) {
+  unwind_state state = {
+    .buffer_cursor = stack_buffer,
+    .buffer_end = stack_buffer + stack_size,
+  };
+
+  _Unwind_Backtrace(UnwindTrace, &state);
+  return state.buffer_cursor - stack_buffer;
+}
+#endif
+
+#if defined(ANDROID) || defined(OS_DARWIN) || (defined(__GLIBC__) && !defined(PLATFORM_TIZEN))
```

Comment:
This condition is no longer correct - the `__GLIBC__` && !TIZEN` is due to the missing `backtrace(2)` I think, so we should be just doing this on Tizen as well and using the emulation above there as well.

Reply:
I'd rather leave this exclusion as-is so it doesn't inadvertently break Tizen. The emulation of `backtrace` works on Android because the NDK provides libunwind. I haven't confirmed Tizen provides the same functionality.

Reply:
Okay, that is fair. I think that we should figure out how to get Tizen into the CI.